go/cmd/vttestserver/vttestserver_test.go

```diff
@@ -243,6 +317,12 @@ func assertEqual(t *testing.T, actual string, expected string, message string) {
 	}
 }
 
-func resetFlags(args []string) {
-	os.Args = args
+func resetFlags() {
+	os.Args = initArgs
+	config = vttest.Config{}
```

Comment:
I was a bit baffled about why this passes the tests because I wasn't sure how the `flag` package knows the new pointers to the new struct 😄 

I'm working on it fixing this in a better way but I think it's fairly independent from the rest of the PR.

The reason I was baffled is because of this:


```
package main

import (
 -"flag"
 -"fmt"
 -"os"
)

type Config struct {
 -Dir  string
 -Test string
}

var config Config

func init() {
 -flag.StringVar(&config.Dir, "dir", "/tmp/default", "...")
 -flag.StringVar(&config.Test, "test", "test-default", "...")
}

func main() {
 -initArgs := append([]string{}, os.Args...)
 -os.Args = append(os.Args, "-test=not-default")
 -flag.Parse()
 -fmt.Printf("config = %+v\n", config)

 -os.Args = initArgs
 -os.Args = append(os.Args, "-dir=/tmp/not-default")
 -config = Config{}
 -flag.Parse()
 -fmt.Printf("config = %+v\n", config)
}
```

The program above produces this output:

```
config = {Dir:/tmp/default Test:not-default}
config = {Dir:/tmp/not-default Test:}  // missing its default value of course
```

I still don't understand how the new `config` gets the same pointer as the old `config`.. Must be some memory optimization in go runtime ❓